assets/js/editor/editor.js

```diff
@@ -38,14 +41,43 @@ class ConnectToBlocksEditorSupport {
 
 		registerPlugin('convert-to-blocks', {
 			render: () => {
-				transformer.execute();
+				// Don't render more than once, to avoid triggering multiple migrations
+				if (loaded) {
+					return null;
+				}
+
+				loaded = true;
+
+				// This delay allows Gutenberg to initialize legacy content into freeform blocks
+				setTimeout(() => {
+					const result = transformer.execute();
+					const config = window.convert_to_blocks_agent || false;
+
+					// if no migration config, then ignore this request
+					if (!config) {
+						return null;
+					}
+
+					const client = new MigrationClient(config);
+
+					// if no blocks transformed, then we can jump to the next post
+					if (!result) {
+						client.next();
+						return null;
+					}
+
+					client.save();
+
+					return null;
+				}, 500);
```

Comment:
Not sure but is it good to determine the fixed delay time? What if more time is required for some posts and those are skipped during the conversion?

Reply:
@faisal-alvi This is to give time to Gutenberg to initialize. My tests showed 100ms is enough. I'm erring on the side of caution and using 500ms. Maybe something to make configurable in the future.